google/cloud/forseti/scanner/scanner.py

```diff
@@ -54,8 +115,13 @@ def run(model_name=None, progress_queue=None, service_config=None):
                 scanner.__class__.__name__)
             progress_queue.put(log_message)
             LOGGER.error(log_message, exc_info=True)
+            failed.append(scanner.__class__.__name__)
+        else:
+            succeeded.append(scanner.__class__.__name__)
     # pylint: enable=bare-except
     log_message = 'Scan completed!'
+    mark_scanner_index_complete(
+        service_config, scanner_index_id, succeeded, failed)
```

Comment:
The logic should determining the run state would be better off in a separate helper method.
```
run_state = determine_run_state(successful_scanners, failed_scanners)
```
This would be much easier to test.  And can simplify the `mark_scanner_index_complete()` method's signature and functionality to a single operation.
```
mark_scanner_index_complete(
    service_config,
    scanner_index_id,
    run_state)
```